### ISSUE TITLE
language-relations dependency and Contao 3.5

### ISSUE THREAD

**Issue by fbender** [terminal42/contao-changelanguage#56]
I will relaunch a Contao 2.9-based project soon. For the relaunch, I want to re-use [changelanguage] again. Since I also want to use [backend-language-switcher], I'd like to have [changelanguage] also depend on [language-relations] to ensure consistency. This is already done in the `develop` branch and the Composer `dev-develop` version. However, that branch has not been updated since last November and the package is not marked as compatible with Contao 3.5. Also, I'm a bit hesitant to run a develop version on a production system. This leads to the following questions:
1. Is the stable version compatible with [backend-language-switcher], both in running the extensions in parallel and the configuration (i.e. do I have to configure the language relation twice or are the same settings used)?
2. Is it safe to use the develop version?
3. Will you make the version that depends on [language-relations] compatible with Contao 3.5? 
4. Will you continue development of the version in `develop` at all, or will you go straight to Contao 4 resp. the branch? 
5. If the latter is true, will that version (the one for Contao 4) also be compatible with Contao 3.5? Will that version switch to [language-relations] as the base?
6. If the answers to No. 3 and No. 5 is no, what is holding you back? How can I help to make that happen?

Thanks in advance!

**Comment by Toflar** [comment]
> Is the stable version compatible with [backend-language-switcher], both in running the extensions in parallel and the configuration (i.e. do I have to configure the language relation twice or are the same settings used)?

I don't know that extension, so I can't tell if they're compatible.

> Is it safe to use the develop version?

No, that's why it is `develop`.

> Will you make the version that depends on [language-relations] compatible with Contao 3.5?

Yes, the reason why the `develop` branch has not been released is simple: language-relations only supports page relations for now. However, changelanguage also provided a hook for other translations such as newslanguage etc. I had several phone calls with @frontendschlampe and we agreed that language-relations must support "types" of relations so you can also link news entries amongst each other, calendar entries and whatever. But this was never implemented in language-relations and it's the only thing hindering me from releasing a new changelanguage version.

 > Will you continue development of the version in develop at all, or will you go straight to Contao 4 resp. the branch?

Yes, I will. As soon as language-relations is ready.

> If the latter is true, will that version (the one for Contao 4) also be compatible with Contao 3.5? Will that version switch to [language-relations] as the base?

Given the fact that Contao 3.5 will have support until 2019 and we use both 3.5 and 4.0, we'll have working versions for both. Both depending on language-relations (which must be Contao 4 compatible too, obviously).

> If the answers to No. 3 and No. 5 is no, what is holding you back? How can I help to make that happen?

Ask @frontendschlampe or his brother :-)

**Comment by frontendschlampe** [comment]
We're very sorry. We know, that we have to do it and we also want to do it. But we don't find the time the last months. But our target will be the end of the year! I also need it for a project in future ...
